lib/crawler.js

```diff
@@ -359,71 +431,173 @@ Crawler.prototype._buildHttpRequest = function _buildHTTPRequest (options) {
 	}
 };
 
-Crawler.prototype._onContent = function _onContent (error, options, response) {
+Crawler.prototype._buildHttp2Session = function _buildHttp2Session(targetHost) {
+	const self = this;
+
+	self.http2Connections[targetHost] = http2.connect(targetHost);
+
+	log('debug', `connect to a new ${targetHost}`);
+	self.http2Connections[targetHost].on('error', (err) => {
+		log('warn', `Http2 stession error ${targetHost}, got error ${err}`);
+	});
+
+	self.http2Connections[targetHost].on('goaway', () => {
+		log('debug', `Http2 session${targetHost} connection goaway`);
+	});
+
+	self.http2Connections[targetHost].on('connect', () => {
+		log('debug', `Http2 session${targetHost} connection init`);
+	});
+
+	self.http2Connections[targetHost].on('timeout', () => {
```

Comment:
we don't set the timeout milliseconds so the session won't emit the `timeout` event according to the document of Nodejs.